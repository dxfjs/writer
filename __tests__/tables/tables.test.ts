import { Seeder, Tables, TagsManager } from "@/index";

describe("Tables class", () => {
  it("should create a tables section", () => {
    const tables = new Tables({ seeder: new Seeder() });
    tables.addLType({
      name: "DASHDOT",
      descriptive: "__ . ",
      elements: [1, 1, -1, 0, -1],
    });
    const mg = new TagsManager();
    tables.tagify(mg);
    expect(mg.stringify()).toMatchSnapshot();
  });
});
