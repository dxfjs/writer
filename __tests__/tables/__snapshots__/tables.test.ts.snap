// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tables class > should create a tables section 1`] = `
"0
SECTION
2
TABLES
0
TABLE
2
APPID
5
1
330
0
100
AcDbSymbolTable
70
1
0
APPID
5
A
330
1
100
AcDbSymbolTableRecord
100
AcDbRegAppTableRecord
2
ACAD
70
0
0
ENDTAB
0
TABLE
2
BLOCK_RECORD
5
2
330
0
100
AcDbSymbolTable
70
2
0
BLOCK_RECORD
5
B
330
0
100
AcDbSymbolTableRecord
100
AcDbBlockTableRecord
2
*Model_Space
340
0
70
0
280
1
281
0
1001
ACAD
1002
{
1002
}
0
BLOCK_RECORD
5
C
330
0
100
AcDbSymbolTableRecord
100
AcDbBlockTableRecord
2
*Paper_Space
340
0
70
0
280
1
281
0
1001
ACAD
1002
{
1002
}
0
ENDTAB
0
TABLE
2
DIMSTYLE
5
3
330
0
100
AcDbSymbolTable
70
1
100
AcDbDimStyleTable
0
DIMSTYLE
105
D
330
3
100
AcDbSymbolTableRecord
100
AcDbDimStyleTableRecord
2
Standard
70
0
340
12
0
ENDTAB
0
TABLE
2
LTYPE
5
5
330
0
100
AcDbSymbolTable
70
4
0
LTYPE
5
F
330
5
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
ByLayer
70
0
3

72
65
73
0
40
0
0
LTYPE
5
10
330
5
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
ByBlock
70
0
3

72
65
73
0
40
0
0
LTYPE
5
11
330
5
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
Continuous
70
0
3
Solid line
72
65
73
0
40
0
0
LTYPE
5
14
330
5
100
AcDbSymbolTableRecord
100
AcDbLinetypeTableRecord
2
DASHDOT
70
0
3
__ . 
72
65
73
5
40
4
49
1
74
0
49
1
74
0
49
-1
74
0
49
0
74
0
49
-1
74
0
0
ENDTAB
0
TABLE
2
STYLE
5
6
330
0
100
AcDbSymbolTable
70
1
0
STYLE
5
12
330
6
100
AcDbSymbolTableRecord
100
AcDbTextStyleTableRecord
2
Standard
70
0
40
0
41
1
50
0
71
0
42
1
3
txt
4

1001
ACAD
1002
{
1071
34
1002
}
0
ENDTAB
0
TABLE
2
LAYER
5
4
330
0
100
AcDbSymbolTable
70
1
0
LAYER
5
E
330
4
100
AcDbSymbolTableRecord
100
AcDbLayerTableRecord
2
0
70
0
62
7
6
Continuous
390
0
0
ENDTAB
0
TABLE
2
VIEW
5
7
330
0
100
AcDbSymbolTable
70
0
0
ENDTAB
0
TABLE
2
UCS
5
8
330
0
100
AcDbSymbolTable
70
0
0
ENDTAB
0
TABLE
2
VPORT
5
9
330
0
100
AcDbSymbolTable
70
1
0
VPORT
5
13
330
9
100
AcDbSymbolTableRecord
100
AcDbViewportTableRecord
2
*Active
70
0
41
1
0
ENDTAB
0
ENDSEC"
`;
